azurerm/azurerm_kubernetes_cluster/simple/main.tf

```diff
@@ -0,0 +1,53 @@
+# Summary: A simple Azure Kubernetes Cluster
+# Explanation: Also known as Azure Kubernetes Service (AKS)
+
+# Documentation: https://www.terraform.io/docs/language/settings/index.html
+terraform {
+  required_version = ">= 0.14.0"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 2.0"
+    }
+  }
+}
+
+# Documentation: https://www.terraform.io/docs/language/values/variables.html
+variable "azure_subscription_id" {
+  type = string
+}
+
+# Documentation: https://www.terraform.io/docs/language/providers/requirements.html
+# Documentation: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs
+provider "azurerm" {
+  features {}
+
+  subscription_id = var.azure_subscription_id
+}
+
+# Resource Group
+# Documentation: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_group
+resource "azurerm_resource_group" "changeme_simple_kubernetes_cluster_resource_group" {
+  name     = "changeme-simple-kubernetes-cluster-resource-group"
+  location = "West Europe"
+}
+
+# Kubernetes Cluster within the Resource Group
+# Documentation: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster
+resource "azurerm_kubernetes_cluster" "changeme_simple_kubernetes_cluster" {
+  name                = "changeme-simple-kubernetes-cluster"
```

Comment:
Apply error caused by name too long:
- https://github.com/ContainerSolutions/terraform-examples/runs/3261013061?check_suite_focus=true#step:5:474
```
╷
│ Error: creating Managed Kubernetes Cluster "changeme-simple-kubernetes-cluster" (Resource Group "changeme-simple-kubernetes-cluster-resource-group"): containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="InvalidParameter" Message="The length of the node resource group name is too long. The maximum length is 80 and the length of the value provided is 98. Please see https://aka.ms/aks-naming-rules for more details." Target="name"
│
│   with azurerm_kubernetes_cluster.changeme_simple_kubernetes_cluster,
│   on main.tf line 37, in resource "azurerm_kubernetes_cluster" "changeme_simple_kubernetes_cluster":
│   37: resource "azurerm_kubernetes_cluster" "changeme_simple_kubernetes_cluster" {
│
╵
```

Reply:
Some more info in docs: https://docs.microsoft.com/en-gb/azure/aks/troubleshooting#what-naming-restrictions-are-enforced-for-aks-resources-and-parameters

Fix for this:  https://github.com/ContainerSolutions/terraform-examples/commit/d5058ecc2d71d480201d1efba6cf4a68c2f42452